src/sql/workbench/contrib/commandLine/electron-sandbox/commandLine.ts

```diff
@@ -118,37 +118,27 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 			});
 			return;
 		}
-		let connectedContext: azdata.ConnectedContext = undefined;
-		if (profile) {
-			if (profile.providerName && !this._capabilitiesService.providers[profile.providerName]) {
-				const installed = await this._connectionManagementService.handleUnsupportedProvider(profile.providerName);
-				if (!installed) {
-					// User cancelled install prompt so exit early since we won't be able to connect
-					return;
+		if (commandName) {
+			if (profile && commandName === Command.connect) {
+				if (this._notificationService) {
+					this._notificationService.status(localize('connectingLabel', "Connecting: {0}", profile.serverName), { hideAfter: 2500 });
+				}
+				try {
+					await this._connectionManagementService.connectIfNotConnected(profile, args.showDashboard ? 'dashboard' : 'connection', true);
+					// Before sending to extensions, we should a) serialize to IConnectionProfile or things will fail,
+					// and b) use the latest version of the profile from the service so most fields are filled in.
+					let updatedProfile = this._connectionManagementService.getConnectionProfileById(profile.id);
+					connectedContext = { connectionProfile: new ConnectionProfile(this._capabilitiesService, updatedProfile).toIConnectionProfile() };
+				} catch (err) {
+					this.logService.warn('Failed to connect due to error: ' + getErrorMessage(err));
 				}
-				// Recreate the profile here now that we have our provider registered so that we get the correct
-				// option names. See https://github.com/microsoft/azuredatastudio/issues/20773 for details about the issue
-				profile = this.readProfileFromArgs(args);
-			}
-			if (this._notificationService) {
-				this._notificationService.status(localize('connectingLabel', "Connecting: {0}", profile.serverName), { hideAfter: 2500 });
-			}
-			try {
-				await this._connectionManagementService.connectIfNotConnected(profile, args.showDashboard ? 'dashboard' : 'connection', true);
-				// Before sending to extensions, we should a) serialize to IConnectionProfile or things will fail,
-				// and b) use the latest version of the profile from the service so most fields are filled in.
-				let updatedProfile = this._connectionManagementService.getConnectionProfileById(profile.id);
-				connectedContext = { connectionProfile: new ConnectionProfile(this._capabilitiesService, updatedProfile).toIConnectionProfile() };
-			} catch (err) {
-				this.logService.warn('Failed to connect due to error: ' + getErrorMessage(err));
 			}
-		}
-		if (commandName) {
 			if (this._notificationService) {
 				this._notificationService.status(localize('runningCommandLabel', "Running command: {0}", commandName), { hideAfter: 2500 });
 			}
-			await this._commandService.executeCommand(commandName, connectedContext);
-		} else if (connectedContext) {
+			await this.runCommandHandler(commandName, args);
```

Comment:
This is meant to allow users to run a command immediately upon startup so we ideally shouldn't be removing that functionality. And I'm not sure about overloading "command" to be both "hardcoded" commands in this file and the registered commands...so maybe we can find a different name for the hardcoded ones (action perhaps?)

I forget exactly why we added that in the first place - it was added in https://github.com/microsoft/azuredatastudio/pull/3690 @shueybubbles do you recall what the exact use case for this was?

Reply:
this is how SSMS opens ADS to the New Notebook experience

From the object explorer XML:
```xml
    <Object name='NewNotebook' base='StartAzureDataStudio'>
      <Properties>
        <LocalizableProperty Name='MenuText'>New &amp;Notebook</LocalizableProperty>
        <Property Name="AdditionalArguments">--command=notebook.command.new</Property>
```

Reply:
Makes sense, thanks for context. Will update PR accordingly.